doc/arkode/guide/source/Usage/SPRKStep_c_interface/User_callable.rst

```diff
@@ -0,0 +1,1462 @@
+
+   :param arkode_mem: pointer to the SPRKStep memory block.
+   :param fp: pointer to use for printing the solver parameters.
+
+   :return:
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the SPRKStep memory was ``NULL``
+
+   **Notes:**
+      The *fp* argument can be ``stdout`` or ``stderr``, or it
+      may point to a specific file created using ``fopen``.
+
+      When run in parallel, only one process should set a non-NULL value
+      for this pointer, since parameters for all processes would be
+      identical.
+
+
+
+.. _ARKODE.Usage.SPRKStep.Reinitialization:
+
+SPRKStep re-initialization function
+-------------------------------------
+
+To reinitialize the SPRKStep module for the solution of a new problem,
+where a prior call to :c:func:`SPRKStepCreate` has been made, the
+user must call the function :c:func:`SPRKStepReInit()`.  The new
+problem must have the same size as the previous one.  This routine
+retains the current settings for all SPRKStep module options and
+performs the same input checking and initializations that are done in
+:c:func:`SPRKStepCreate`, but it performs no memory allocation as is
+assumes that the existing internal memory is sufficient for the new
+problem.  A call to this re-initialization routine deletes the
+solution history that was stored internally during the previous
+integration, and deletes any previously-set *tstop* value specified via a
+call to :c:func:`SPRKStepSetStopTime()`.  Following a successful call to
+:c:func:`SPRKStepReInit()`, call :c:func:`SPRKStepEvolve()` again for the
+solution of the new problem.
+
+The use of :c:func:`SPRKStepReInit()` requires that the number of
+Runge--Kutta stages, denoted by *s*, be no larger for the new problem than
+for the previous problem.  This condition is automatically fulfilled
+if the method order *q* is left unchanged.
+
+One important use of the :c:func:`SPRKStepReInit()` function is in the
+treating of jump discontinuities in the RHS function.  Except in cases
+of fairly small jumps, it is usually more efficient to stop at each
+point of discontinuity and restart the integrator with a readjusted
+ODE model, using a call to this routine.  To stop when the location
+of the discontinuity is known, simply make that location a value of
+``tout``.  To stop when the location of the discontinuity is
+determined by the solution, use the rootfinding feature.  In either
+case, it is critical that the RHS function *not* incorporate the
+discontinuity, but rather have a smooth extension over the
+discontinuity, so that the step across it (and subsequent rootfinding,
+if used) can be done efficiently.  Then use a switch within the RHS
+function (communicated through ``user_data``) that can be flipped
+between the stopping of the integration and the restart, so that the
+restarted problem uses the new values (which have jumped).  Similar
+comments apply if there is to be a jump in the dependent variable
+vector.
```

Comment:
A brief literature review (mainly a quick read of [this paper](https://www.sciencedirect.com/science/article/pii/S0021999121007415?casa_token=B-3PdJGAnYwAAAAA:irVYQ5VxxDyTLhdEDvTakvQEV7MqlEqw9WppZVLKt-tqeqPKIexe1nijYeAdvzSCv2xEALHrbg)) seems to indicate that jump discontinuities could occur in Hamiltonian problems that are physically interesting. However, its less clear how well (if at all) this approach would work for these problems. SHould we leave the comment in for now and then maybe do an experiment in the future? Or should we leave the comment out (and still do a future experiment)?

Reply:
Since we are less confident about this use case, then perhaps we can leave it in with some caveats, and include a reference to that citation.  Perhaps something like:

One potential use of the :c:func:`SPRKStepReInit()` function is to treat jump discontinuities in the RHS function [citation].  In lieu of including `if` statements within the RHS function to handle discontinuities, it may be more computationally efficient to stop at each point of discontinuity (e.g., through use of ``tout`` or the rootfinding feature) and restart the integrator with a readjusted ODE model, using a call to this routine.  We note that for the solution to retain temporal accuracy, the RHS function should *not* incorporate the discontinuity.